libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestClient.java

```diff
@@ -650,8 +650,12 @@ private void refreshAccessToken(HttpResponse response) throws IOException {
 				if (newAuthToken != null) {
 					setAuthToken(newAuthToken);
 				}
+
                 // Check if the instanceUrl changed
                 String instanceUrl = authTokenProvider.getInstanceUrl();
+				if (instanceUrl == null) {
```

Comment:
When refresh token is revoked, ```instanceUrl``` is guaranteed to be ```null``` here. We try to URI parse ```instanceUrl```, which breaks. We didn't notice it on our sample apps because sample apps use ```sendAsync```, which bubbles up the error, whereas ```sendSync``` doesn't.